src/client/nosto.d.ts

```diff
@@ -4072,4 +4058,112 @@ declare const api: {
+                    campaignId?: string;
+                    popupId?: string;
+                    effect: Partial<PopupEffect>;
+                    trigger: PopupTrigger;
+                    preview?: boolean;
+                    cart?: PopupCart;
+                }): void;
+                close(): void;
+            };
+            preview: (popupId: string, campaignId: string, effect?: PopupEffect) => void;
+            previewById: (popupId: string, effect: PopupEffect) => void;
+            open: (popupId: string, response: ResponseData | null, effect: PopupEffect, trigger: PopupTrigger) => void;
+            okToOpen: (popupId: string, condition: Condition, responseData?: ResponseData) => boolean;
+            openCheck: (popupId: string, condition?: Condition, responseData?: ResponseData) => string | null;
+            stampOnCheckoutPage: () => void;
+            openMinimized: () => void;
+            done: (popupId: string) => void;
+            writePopupAttribute: typeof writePopupAttribute;
+            readPopupAttributes: typeof readPopupAttributes;
+        } | null;
+        preview(popupId: string, campaignId: string, effect?: PopupEffect): {
+            internal: {
+                showPopup(showPopupOptions: {
+                    campaignId?: string;
+                    popupId?: string;
+                    effect: Partial<PopupEffect>;
+                    trigger: PopupTrigger;
+                    preview?: boolean;
+                    cart?: PopupCart;
+                }): void;
+                close(): void;
+            };
+            preview: (popupId: string, campaignId: string, effect?: PopupEffect) => void;
+            previewById: (popupId: string, effect: PopupEffect) => void;
+            open: (popupId: string, response: ResponseData | null, effect: PopupEffect, trigger: PopupTrigger) => void;
+            okToOpen: (popupId: string, condition: Condition, responseData?: ResponseData) => boolean;
+            openCheck: (popupId: string, condition?: Condition, responseData?: ResponseData) => string | null;
+            stampOnCheckoutPage: () => void;
+            openMinimized: () => void;
+            done: (popupId: string) => void;
+            writePopupAttribute: typeof writePopupAttribute;
+            readPopupAttributes: typeof readPopupAttributes;
+        };
+    };
+} | {
+    activate(): void;
+    campaignList(): (PopupTriggerSettingsDTO & {
+        type?: PopupTrigger;
+    })[];
+    openPopup(): void;
+    enablePopup(): void;
+    disablePopup(): void;
+    setTriggers(): void;
+    discountPopup: {
+        instance: {};
+        preview(): void;
+    };
+    sortedCampaignsWithType?: undefined;
+};
+
+/**
+ * @hidden
+ */
+type Overlay = ReturnType<typeof createOverlay>;
+
+interface Store {
+    getCustomerId(): Maybe<string>;
+    setCustomerId(id: string): void;
+}
+
+/** *****************************************************************************
```

Comment:
we should also try to get rid of these, either in nosto-js or client script d.ts export
they sometimes leak into the typdoc docs

Reply:
All removed.